### PR TITLE
Plug in the Authorization Module in keyserver.go

### PR DESCRIPTION
**Pull request by cesarghali**
This PR plugs in the authorization module implemented in #698 in `keyserver.go`. This requires changing the authentication module to produce an `*authentication.SecurityContext` object that is passed to the `IsAuthorized` API.

### CODE REVIEW

File: cmd/keytransparency-server/main.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ import (
 	"github.com/google/trillian"
 
 	"github.com/golang/glog"
-	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
```
Comment (gdbelvin) [inline cmd/keytransparency-server/main.go]: it would be nice to not rename this import if possible

File: cmd/keytransparency-server/main.go [hunk 2]
```diff
@@ -169,6 +170,7 @@ func main() {
 	default:
 		glog.Exitf("Invalid auth-type parameter: %v.", *authType)
 	}
+	authz := authorization.New()
```
Comment (gdbelvin) [inline cmd/keytransparency-server/main.go]: `New()` Uses an empty policy proto right?  aka, default authorization

File: core/keyserver/keyserver.go [hunk 3]
```diff
@@ -229,17 +234,22 @@ func (s *Server) ListEntryHistory(ctx context.Context, in *tpb.ListEntryHistoryR
...
-	case authentication.ErrWrongUser:
-		return nil, grpc.Errorf(codes.PermissionDenied, "Permission denied")
 	case authentication.ErrMissingAuth:
 		return nil, grpc.Errorf(codes.Unauthenticated, "Missing authentication header")
 	default:
 		glog.Warningf("Auth failed: %v", err)
 		return nil, grpc.Errorf(codes.Unauthenticated, "Unauthenticated")
 	}
+	// Validate proper authorization.
+	// TODO: use the correct appID.
```
Comment (gdbelvin) [inline core/keyserver/keyserver.go]: AppID is present in the user request `in.AppId`

File: cmd/keytransparency-server/main.go (hunk 2 above)
Comment (cesarghali) [inline cmd/keytransparency-server/main.go]: Yes, that's correct.

File: core/keyserver/keyserver.go (hunk 3 above)
Comment (cesarghali) [inline core/keyserver/keyserver.go]: Oh, I missed it somehow, thanks for the tip.

**Pull request merged**